environment/src/main/java/com/sequenceiq/environment/environment/flow/stop/handler/StopFreeIpaHandler.java

```diff
@@ -43,12 +43,16 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
         EnvironmentDto environmentDto = environmentDtoEvent.getData();
         try {
             if (Strings.isNullOrEmpty(environmentDto.getParentEnvironmentCrn())) {
-                freeIpaService.describe(environmentDto.getResourceCrn()).ifPresent(freeIpa -> {
-                    if (freeIpa.getStatus() != null && !freeIpa.getStatus().isStoppable()) {
+                freeIpaService.describe(environmentDto.getResourceCrn()).ifPresentOrElse(freeIpa -> {
+                    if (freeIpa.getStatus() != null && (freeIpa.getStatus().isStoppedPhase() || freeIpa.getStatus().isStopInProgressPhase())) {
+                        LOGGER.info("Freeipa is already {}, so it won't started", freeIpa.getStatus());
```

Comment:
Grammar: `it won't be started` I think it should be something like `Stop has already been trigger continuing without new stop trigger.

Reply:
I didn't want to add a new level to here

Reply:
This message does not contain the status. I think it is a useful information

Reply:
I meant only the last sentence which was faulty.